inc/commonitilobject.class.php

```diff
@@ -6356,12 +6380,20 @@ static function commonListHeader($output_type = Search::HTML_OUTPUT, $mass_id =
       $items[__('Priority')]           = "priority";
       $items[_n('Requester', 'Requesters', 1)]          = "users_id";
       $items[__('Assigned')]           = "users_id_assign";
-      if (static::getType() == 'Ticket') {
-         $items[_n('Associated element', 'Associated elements', Session::getPluralNumber())] = "";
+
+      if (!$ticket_stats) {
+         if (static::getType() == 'Ticket') {
+            $items[_n('Associated element', 'Associated elements', Session::getPluralNumber())] = "";
+         }
+         $items[__('Category')]           = "glpi_itilcategories.completename";
+         $items[__('Title')]              = "name";
+         $items[__('Planification')]      = "glpi_tickettasks.begin";
+      } else {
+         $items[__('Title')]             = "name";
+         $items[__('Take into account')] = "takeintoaccount_delay_stat";
+         $items[__('Resolution')]        = "solve_delay_stat";
+         $items[__('Pending')]           = "waiting_duration";
       }
```

Comment:
It is really hard to read/understand the code that produce the list and its headers, and I guess it will be hard to maintain it.
Maybe `$ticket_stats` value should only act on corresponding columns display.

Reply:
I think I understand what you mean, this issue is that we need to remove some fields to make space (category, planification) but keep the field in the middle of them (title).
I actually think checking column by column would end up more messy.

IMO the real issue is that this method is very hard to override in it's current state and force us to use the $params array to deal with custom code.
Maybe the method should be split in two: one part should be in commondbtm and handle the "framework" (calls to `showNewLine` / `showHeaderItem`) with a new `fields` parameter that would contain the labels to display.
The default fields could be placed in another method.
Do you see what I mean ?